### PR TITLE
feat: Extract Alert Engine WS connector from the main repository

### PR DESCRIPTION
**Pull request by brasseld**
closes gravitee-io/issues#7778

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `1.0.0-issues-7778-extract-ae-connector-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/ae/gravitee-alert-engine-connectors/1.0.0-issues-7778-extract-ae-connector-SNAPSHOT/gravitee-alert-engine-connectors-1.0.0-issues-7778-extract-ae-connector-SNAPSHOT.zip)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    <name>Gravitee.io - Alert Engine - Connectors</name>
+
+    <modules>
+        <module>gravitee-alert-engine-connector-api</module>
+        <module>gravitee-alert-engine-connector-core</module>
+        <module>gravitee-alert-engine-connector-ws</module>
+    </modules>
+
+    <properties>
+        <gravitee-bom.version>2.4</gravitee-bom.version>
```
Comment (phiz71) [inline pom.xml:43]: prefer use 2.5, it fixes a bug that occurs during release

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+        <gravitee-node.version>1.24.0</gravitee-node.version>
+        <gravitee-plugin.version>1.23.1</gravitee-plugin.version>
+        <gravitee-alert-api.version>1.9.0</gravitee-alert-api.version>
+        <mockito.version>4.5.1</mockito.version>
+        <junit-jupiter.version>5.8.2</junit-jupiter.version>
```
Comment (phiz71) [inline pom.xml:48]: junit-jupiter is imported in bom 2.5, you don't need to specify its version here

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+                <version>${gravitee-node.version}</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+
+            <!-- Test dependencies -->
+            <dependency>
+                <groupId>org.junit.jupiter</groupId>
+                <artifactId>junit-jupiter-engine</artifactId>
+                <version>${junit-jupiter.version}</version>
```
Comment (phiz71) [inline pom.xml:93]: junit-jupiter is imported in bom 2.5, you don't need to specify its version here

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+                <scope>test</scope>
+            </dependency>
+            <dependency>
+                <groupId>org.junit.jupiter</groupId>
+                <artifactId>junit-jupiter-api</artifactId>
+                <version>${junit-jupiter.version}</version>
```
Comment (phiz71) [inline pom.xml:99]: junit-jupiter is imported in bom 2.5, you don't need to specify its version here

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+                <scope>test</scope>
+            </dependency>
+            <dependency>
+                <groupId>org.junit.jupiter</groupId>
+                <artifactId>junit-jupiter-params</artifactId>
+                <version>${junit-jupiter.version}</version>
```
Comment (phiz71) [inline pom.xml:105]: junit-jupiter is imported in bom 2.5, you don't need to specify its version here

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+                    <target>11</target>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>com.hubspot.maven.plugins</groupId>
+                <artifactId>prettier-maven-plugin</artifactId>
+                <version>0.17</version>
+                <configuration>
+                    <nodeVersion>12.13.0</nodeVersion>
+                    <prettierJavaVersion>1.0.1</prettierJavaVersion>
```
Comment (phiz71) [inline pom.xml:204]:
```suggestion
                    <prettierJavaVersion>1.6.1</prettierJavaVersion>
```

File: gravitee-alert-engine-connector-ws/pom.xml [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+        <artifactId>gravitee-alert-engine-connectors</artifactId>
+        <version>1.0.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>gravitee-alert-engine-connectors-ws</artifactId>
+    <name>Gravitee.io - Alert Engine - Connectors - Websocket Support</name>
+
+    <properties>
+        <!-- Property used by the publication job in CI-->
+        <publish-folder-path>graviteeio-ae/plugins/connectors/</publish-folder-path>
```
Comment (phiz71) [inline gravitee-alert-engine-connector-ws/pom.xml:35]:
```suggestion
        <publish-folder-path>graviteeio-ae/plugins/connectors</publish-folder-path>
```

File: gravitee-alert-engine-connector-ws/pom.xml [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+            <groupId>io.gravitee.node</groupId>
+            <artifactId>gravitee-node-management</artifactId>
+            <scope>provided</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>javax.annotation</groupId>
+            <artifactId>javax.annotation-api</artifactId>
+            <version>1.3.2</version>
+        </dependency>
```
Comment (phiz71) [inline gravitee-alert-engine-connector-ws/pom.xml:61]: not provided ?

**Comment by phiz71** [comment]
I could review the pom and distribution part, but need someone else to review the code

File: gravitee-alert-engine-connector-ws/pom.xml (hunk 8 above)
Comment (brasseld) [reply gravitee-alert-engine-connector-ws/pom.xml:61]: already provided by the node

File: gravitee-alert-engine-connector-ws/src/main/resources/plugin.properties [hunk 9]
```diff
@@ -0,0 +1,6 @@
+id=alert-engine-connector-ws
+name=${project.name}
+version=${project.version}
+description=${project.description}
+class=com.graviteesource.ae.connector.ws.WsConnectorPlugin
```
Comment (phiz71) [inline gravitee-alert-engine-connector-ws/src/main/resources/plugin.properties:5]:
```suggestion
class=io.gravitee.ae.connector.ws.WsConnectorPlugin
```

Comment (phiz71) [reply gravitee-alert-engine-connector-ws/src/main/resources/plugin.properties]: I pushed the fix

**Comment by phiz71** [comment]
I've tested locally. It's ok for me

**Pull request merged**